src/test/java/edu/ucsb/cs56/ucsb_open_lab_scheduler/controllers/NavbarTest.java

```diff
@@ -0,0 +1,149 @@
+package edu.ucsb.cs56.ucsb_open_lab_scheduler.controllers;
+
+import edu.ucsb.cs56.ucsb_open_lab_scheduler.advice.AuthControllerAdvice;
+import edu.ucsb.cs56.ucsb_open_lab_scheduler.repositories.*;
+import edu.ucsb.cs56.ucsb_open_lab_scheduler.services.MembershipService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(ApplicationController.class)
+public class NavbarTest {
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private CourseOfferingRepository cor;
+
+    @MockBean
+    private TutorAssignmentRepository tar;
+
+    @MockBean
+    private TimeSlotAssignmentRepository tsar;
+
+    @MockBean
+    private ClientRegistrationRepository crr;
+
+    @MockBean
+    private TutorRepository tr;
+
+    @MockBean
+    private AuthControllerAdvice aca;
+
+    @MockBean
+    private MembershipService ms;
+
+    @MockBean
+    private UserRepository aur;
+
+    @MockBean
+    private AdminRepository ar;
+
+
+    /**
+     * This tests the authenticated "guest" user to make sure there is no Google login button in the menu bar,
+     * but instead has the user's name with (Guest) and the proper logout button.
+     * Also tests to make sure Guests do not have links to admin/student resources
+     * @throws Exception
+     */
+    @Test
+    public void testGuestNavbar() throws Exception {
+        when(aca.getFirstName(any())).thenReturn("Joe");
+        when(aca.getLastName(any())).thenReturn("Gaucho");
+        when(aca.getEmail(any())).thenReturn("joegaucho@ucsb.edu");
+        when(aca.getRole(any())).thenReturn("Guest");
+        when(aca.getIsLoggedIn(any())).thenReturn(true);
+        // Now check that the name in the header is Joe
+        mvc.perform(MockMvcRequestBuilders.get("/").accept(MediaType.TEXT_HTML)).andExpect(status().isOk())
+                .andExpect(content().contentType("text/html;charset=UTF-8"))
+                .andExpect(xpath("/html/body/div[@class='container']/nav[@class='navbar navbar-expand-lg navbar-light bg-light']/div[@id='navbarTogglerDemo03']/ul[@class='nav navbar-nav navbar-right']/li[1]/a")
```

Comment:
It might be better to try the least specific selector that will find the content.  Otherwise, the tests become brittle ... i.e. any tiny change to the HTML causes the test to break.   At some, you are just writing the code twice...